[IMG-178] Linux vm support policy (#8481)
circleci/circleci-docs: jekyll/_cci2/linux-vm-support-policy.adoc

---
contentTags:
  platform:
  - Cloud
---
= CircleCI Linux VM images support policy
:page-layout: classic-docs
:page-liquid:
:page-description: CircleCI Linux VM images release, update, and deprecation policy
:icons: font
:experimental:

[#overview]
== Overview

> This document outlines the CircleCI Linux VM (Machine) Image release, update and deprecation policy. This policy applies to all CircleCI Linux VM (Machine) Images built for the Machine executor. Note that there is a separate [CUDA image support policy](https://circleci.com/docs/linux-cuda-images-support-policy/).

Reviewer: Please fix link format

When critical CVEs are disclosed that affect the versions of the operating system or software stack in Linux VM images, we will investigate the impact that this has on our images being used within the CircleCI execution environment.  If customers are impacted by these CVEs we will push a patch fix to the released image(s), this image will supersede the original image.

[#bug-reports-issues-and-prs]
== Bug reports, issues, and PRs

You can file a link:https://support.circleci.com/hc/en-us/requests/new[support ticket] with CircleCI Support for any issues or bugs found with the Linux VM images. Our support team will be able to escalate issues internally to the correct engineering team and provide updates on the issue.

[#image-lifespan-eol]
== Image lifespan / EOL

Image Lifespan will generally follow the support status of Ubuntu by Canonical. Once a new Ubuntu LTS is released, we will begin the deprecation process of the oldest version that we support and schedule it to be removed. This allows us to maintain our aim of two LTS Ubuntu versions effectively.

Current Deprecation (Jan 2024):

[.table.table-striped]
[cols=2*, options="header", stripes=even]
|===
| Version
| Support

> 

Reviewer: Missing Ubuntu 22.04 LTS

| Ubuntu 20.04 LTS
| Build quarterly images for current year and retain Q4 (October) release from past two years.
|===

When Ubuntu 24.04 LTS is released:

[.table.table-striped]
[cols=2*, options="header", stripes=even]
|===
| Version
| Support

| Ubuntu 20.04 LTS
| Deprecated and removed entirely

| Ubuntu 22.04 LTS
| Retain only Q4 (October) releases from past two years

| Ubuntu 24.04 LTS

> | Quarterly images biuld and only Q4 (October) image retained from each year

Reviewer: spelling mistake in biuld -> build

---
contentTags:
  platform:
  - Cloud

>   - Server v4.x

Reviewer: Same server issue/question here

  - Server v3.x
---
= CircleCI Linux VM images support policy
:page-layout: classic-docs
:page-liquid:
:page-description: CircleCI Linux VM images release, update, and deprecation policy
:icons: font
:experimental:

[#overview]
== Overview

> This document outlines the CircleCI Linux VM (Machine) Image release, update and deprecation policy. This policy applies to all CircleCI Linux VM (Machine) Images built for the Machine executor. Note that there is a separate [CUDA image support policy](https://circleci.com/docs/linux-cuda-images-support-policy/).

Reviewer: This does not align with what we got approved

Author: i think it is important to include references and backlinks to a very similar image as this eliminates confusion for all parties and is part of the PR template checklist. I can bring this up with support or maybe Sebastian can chime in here?

Reviewer: I am ok if @sebastian-lerner is ok for this one

Reviewer 2: sorry, what's the confusion? what did get approved and what's the delta?

Author: @sebastian-lerner: the only delta from what was approved is the the addition of this line to differentiate this from the Linux CUDA image

`Note that there is a separate [CUDA image support policy](https://circleci.com/docs/linux-cuda-images-support-policy/).`

Reviewer 2: okay seems fine to me

[#release-policy]
== Release policy

The CircleCI Linux VM images are all based on Ubuntu, we support both ARM64 and AMD64 architectures.

We aim to maintain, support, and release the two most recent LTS versions of Ubuntu.

[#tagging]
== Tagging

Tags we support for this image (tag is what is specified in .circleci/config.yml when requesting an image):

> - `default`: This image is the current stable image. Jobs will default to this if no image or tag is specified.

Reviewer: Differs from approved

Author: will do

- `edge`: This tag is reserved for previews of new releases, which will initially point to this tag. `edge` tags may include incremental updates to the current quarterly (every 3 months) image release, which may change without notice, and is not recommended to be used for production CI workloads. `current` will be updated with these changes after a period of stability.

- Date based tagging: The Ubuntu LTS based images have the image slug format of: ubuntu-<LTS version>:<year>.<month>.<patch>, such as ubuntu-2204:2023.10.1. We recommend using the default version and not pinning to a date version.

We always aim to release a version of this image every 3 months.

[#critical-cve-patches]
== Critical CVE patches

When critical CVEs are disclosed that affect the versions of the operating system or software stack in our remote Docker images, we will investigate the impact that this has on our images being used within the CircleCI execution environment.  If customers are impacted by these CVEs we will push a patch fix to the released image(s), this image will supersede the original image.

[#bug-reports-issues-and-prs]
== Bug reports, issues, and PRs

> You can file a link:https://support.circleci.com/hc/en-us/requests/new[support ticket] with CircleCI Support for any issues or bugs found with the remote Docker images. Our support team will be able to escalate issues internally to the correct engineering team and provide updates on the issue.

Reviewer: This is not remote docker image

[#image-lifespan-eol]
== Image lifespan / EOL

> mage Lifespan will generally follow the support status of Ubuntu by Canonical. Once a new Ubuntu LTS is released, we will begin the deprecation process of the oldest version that we support and schedule it to be removed. This allows us to maintain our aim of two LTS Ubuntu versions effectively.

Author: will do

Current Deprecation (Jan 2024):

[.table.table-striped]
[cols=2*, options="header", stripes=even]
|===
| Version
| Support

| Ubuntu 20.04 LTS
| Build quarterly images for current year and retain Q4 (October) release from past two years.
|===

When Ubuntu 24.04 LTS is released:

[.table.table-striped]
[cols=2*, options="header", stripes=even]
|===
| Version
| Support

| Ubuntu 20.04 LTS

> | Deprecated and removed

Reviewer: Deprecated and removed entirely

| Ubuntu 22.04 LTS
| Retain only Q4 (October) releases from past two years

| Ubuntu 24.04 LTS

> | Build quarterly images for current year and retain Q4 (October) release

Author: will do

---
contentTags:
  platform:
  - Cloud
---
= CircleCI Linux VM images support policy
:page-layout: classic-docs
:page-liquid:
:page-description: CircleCI Linux VM images release, update, and deprecation policy
:icons: font
:experimental:

[#overview]
== Overview

> This document outlines the CircleCI Linux VM (Machine) Image release, update and deprecation policy. This policy applies to all CircleCI Linux VM (Machine) Images built for the Machine executor. Note that there is a separate link:https://circleci.com/docs/linux-cuda-images-support-policy/[CUDA image support policy].

Reviewer 3: ```suggestion
This document outlines the CircleCI Linux VM (machine) image release, update and deprecation policy. This policy applies to all CircleCI Linux VM (machine) images built for the machine executor. Note that there is a separate xref:linux-cuda-images-support-policy#[CUDA image support policy].
```

[#release-policy]
== Release policy

The CircleCI Linux VM images are all based on Ubuntu, we support both ARM64 and AMD64 architectures.

We aim to maintain, support, and release the two most recent LTS versions of Ubuntu.

[#tagging]
== Tagging

> Tags we support for this image (tag is what is specified in .circleci/config.yml when requesting an image):

Reviewer 3: ```suggestion
Tags we support for this image (tag is what is specified in `.circleci/config.yml` when requesting an image):
```

- `default` version is what you will receive if do not specify an image.

> - `current` version of image will receive updates approximately every 3 months.

Reviewer 3: ```suggestion
- `current` version of the image will receive updates approximately every three months.
```

> - `previous` version of image will receive the previous version of current.

Reviewer 3: ```suggestion
- `previous` version of the image will receive the previous version of `current`.
```

> - `edge` tags are reserved for previews of new releases, which will initially point to this tag. The edge tags may include incremental updates to the current quarterly (every 3 months) image release, which may change without notice, and is not recommended to be used for production CI workloads. currentwill be updated with these changes after a period of stability.

Reviewer 3: ```suggestion
- `edge` tags are reserved for previews of new releases, which will initially point to this tag. The edge tags may include incremental updates to the current quarterly (every 3 months) image release, which may change without notice, and is not recommended to be used for production CI workloads. `current` will be updated with these changes after a period of stability.
```

> - `Date based tagging`: The Ubuntu LTS based images have the image slug format of: `ubuntu-<LTS version>:<year>.<month>.<patch>`, such as `ubuntu-2204:2023.10.1`. We recommend using the `default` version and not pinning to a date version.

Reviewer 3: ```suggestion
- Date based tagging: The Ubuntu LTS based images have the image slug format of: `ubuntu-<LTS version>:<year>.<month>.<patch>`, such as `ubuntu-2204:2023.10.1`. We recommend using the `default` version and not pinning to a date version.
```

> We always aim to release a version of this image every 3 months.

Reviewer 3: ```suggestion
We always aim to release a version of this image every three months.
```

[#critical-cve-patches]
== Critical CVE patches

> When critical CVEs are disclosed that affect the versions of the operating system or software stack in Linux VM images, we will investigate the impact that this has on our images being used within the CircleCI execution environment.  If customers are impacted by these CVEs we will push a patch fix to the released image(s), this image will supersede the original image.

Reviewer 3: ```suggestion
When critical CVEs are disclosed that affect the versions of the operating system or software stack in Linux VM images, we will investigate the impact that this has on our images being used within the CircleCI execution environment.  If customers are impacted by these CVEs we will push a patch fix to the released image(s). This image will supersede the original image.
```

[#bug-reports-issues-and-prs]
== Bug reports, issues, and PRs

You can file a link:https://support.circleci.com/hc/en-us/requests/new[support ticket] with CircleCI Support for any issues or bugs found with the Linux VM images. Our support team will be able to escalate issues internally to the correct engineering team and provide updates on the issue.

[#image-lifespan-eol]
== Image lifespan / EOL

Image Lifespan will generally follow the support status of Ubuntu by Canonical. Once a new Ubuntu LTS is released, we will begin the deprecation process of the oldest version that we support and schedule it to be removed. This allows us to maintain our aim of two LTS Ubuntu versions effectively.

Current Deprecation (Jan 2024):

[.table.table-striped]
[cols=2*, options="header", stripes=even]
|===
| Version
| Support

| Ubuntu 20.04 LTS

> | Build quarterly images for current year and retain Q4 (October) release from past two years.

Reviewer 3: ```suggestion
| Build quarterly images for current year and retain Q4 (October) release from the past two years.
```

| Ubuntu 22.04 LTS
| Quarterly images built and only Q4 image retained each year
|===

> When Ubuntu 24.04 LTS is released:

Reviewer 3: ```suggestion
Example: When Ubuntu 24.04 LTS is released
```

[.table.table-striped]
[cols=2*, options="header", stripes=even]
|===
| Version
| Support

| Ubuntu 20.04 LTS
| Deprecated and removed entirely

| Ubuntu 22.04 LTS
| Retain only Q4 (October) releases from past two years

| Ubuntu 24.04 LTS
| Quarterly images built and only Q4 (October) image retained from each year
|===

When an image is selected for deprecation and removal, we will create an announcement on our Discuss forum, along with reaching out via email to developers who have requested one of the deprecated images in their recent jobs.

> We will also plan brownouts to ensure users are aware of the approaching removal of deprecated images. Generally, we will aim to start an EOL process within 3 months of a new version release

Reviewer 3: ```suggestion
We will also plan brownouts to ensure users are aware of the approaching removal of deprecated images. Generally, we will aim to start an EOL process within 3 months of a new version release.
```
